docs(alerts): add more context and quick start guide (#481)
cloudoperators/greenhouse-extensions: alerts/README.md

title: Alerts
---

Learn more about the **alerts** plugin. Use it to activate Prometheus alert management for your Greenhouse organisation. 

The main terminologies used in this document can be found in [core-concepts](https://cloudoperators.github.io/greenhouse/docs/getting-started/core-concepts).

## Overview

> This Plugin includes a preconfigured _Prometheus Alertmanager_, which is deployed and managed via the _Prometheus_ Operator, and _Supernova_, an advanced user interface for Prometheus Alertmanager. In addition, certificates are automatically generated to ensure _Prometheus_ authentication sending alerts to the Alertmanager as well as Slack notification templates are provided.

Reviewer: ```suggestion
This Plugin includes a preconfigured [_Prometheus Alertmanager_](https://prometheus.io/docs/alerting/latest/alertmanager/), which is deployed and managed via the [_Prometheus_ Operator](https://github.com/prometheus-operator/prometheus-operator), and [_Supernova_](https://github.com/cloudoperators/juno/tree/main/apps/supernova), an advanced user interface for Prometheus Alertmanager. Certificates are automatically generated to enable sending alerts from _Prometheus_ to Alertmanager. These alerts can too be sent as Slack notifications with a provided set of notification templates.
```

Components included in this Plugin:

- [Prometheus Alertmanager](https://prometheus.io/docs/alerting/alertmanager)
- [Supernova](https://github.com/cloudoperators/juno/tree/main/apps/supernova)

> Optionally, [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator) can be used to manage _Prometheus_ and _Alertmanager_ instances. Usually, **kube-monitoring** Plugin installs _Prometheus Operator_.

Reviewer: ```suggestion
Optionally, [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator) can be used to manage _Prometheus_ and _Alertmanager_ instances. Usually, **kube-monitoring** Plugin installs _Prometheus Operator_.
```

The config says this installs Prometheus Operator:
https://github.com/cloudoperators/greenhouse-extensions/blob/main/alerts/charts/values.yaml#L104-L106

Should we put in an either or? Either we have kube-monitoring installing this, or - in the unlikely case of not having kube-monitoring this option should be set. 

What happens if we deploy both with operator enabled?

Author: It is disabled again in the alerts `plugindefinition.yaml` as we intend to install it with kube-monitoring. We currently have no solution to this problem other than this documentation. There is a issue in our backlog to track - https://github.com/cloudoperators/greenhouse-extensions/issues/140

Reviewer: Ok, rewriting it.

```suggestion

This Plugin usually is deployed along the **kube-monitoring** Plugin and does not deploy the [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator) itself. However, if you are intending to use it stand-alone, you need to explicitly enable the deployment of Prometheus Operator, otherwise it will not work. It can be done in the configuration interface of the plugin.

Reviewer: got lost, again here:

```suggestion
This Plugin usually is deployed along the **kube-monitoring** Plugin and does not deploy the [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator) itself. However, if you are intending to use it stand-alone, you need to explicitly enable the deployment of Prometheus Operator, otherwise it will not work. It can be done in the configuration interface of the plugin.
```

> ## Note

Reviewer: ```suggestion
## Disclaimer
```

> It is not meant to be a comprehensive package that covers all scenarios. If you are an expert, feel free to configure the plugin according to your needs.

Reviewer: ```suggestion
This is not meant to be a comprehensive package that covers all scenarios. If you are an expert, feel free to configure the plugin according to your needs.
```

The latter could be written to any package, no? :D

Author: It should be already. 🙂

> The Plugin is a depth configured [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/README.md) Helm chart which helps to keep track of versions and community updates.

Reviewer: ```suggestion
The Plugin is a deeply configured [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/README.md) Helm chart which helps to keep track of versions and community updates.
```

What would you like to transport here? Not sure what `depth configured` means here.

Author: The `kube-prometheus stack` is a generic Helm chart that needs to be tailored to our needs. That's what I was trying to convey.

Contribution is highly appreciated. If you discover bugs or want to add functionality to the plugin, then pull requests are always welcome.

## Quick start

This guide provides a quick and straightforward way to use **alerts** as a Greenhouse Plugin on your Kubernetes cluster.

**Prerequisites**

- A running and Greenhouse-onboarded Kubernetes cluster. If you don't have one, follow the [Cluster onboarding](https://cloudoperators.github.io/greenhouse/docs/user-guides/cluster/onboarding) guide.

**Step 1:**

You can install the `alerts` package in your cluster by installing it with [Helm](https://helm.sh/docs/helm/helm_install) manually or let the Greenhouse platform lifecycle it for you automatically. For the latter, you can either:
  1. Go to Greenhouse dashboard and select the **Alerts** Plugin from the catalog. Specify the cluster and required option values.
  2. Create and specify a `Plugin` resource in your Greenhouse central cluster according to the [examples](#examples).

**Step 2:**

After the installation, you can access the **Supernova** UI by navigating to the `Alerts` tab in the Greenhouse dashboard. 

**Step 3:**

> Greenhouse regularly performs integration tests that are bundled with **alerts**. These provide feedback on whether all the necessary resources are installed and continuously up and running. You will find messages about this in the plugin status and also in the Greenhouse dashboard.

Reviewer 2: <3

title: Alerts
---

> Learn more about the **alerts** plugin. Use it to activate Prometheus alert management for your Greenhouse organisation. 

Reviewer: ```suggestion
Learn more about the **alerts** plugin. Use it to activate Prometheus alert management for your Greenhouse organisation.
```

## Disclaimer

This is not meant to be a comprehensive package that covers all scenarios. If you are an expert, feel free to configure the plugin according to your needs.

The Plugin is a deeply configured [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/README.md) Helm chart which helps to keep track of versions and community updates.

It is intended as a platform that can be extended by following the [guide](#managing-alertmanager-configuration). 

Contribution is highly appreciated. If you discover bugs or want to add functionality to the plugin, then pull requests are always welcome.

## Quick start

This guide provides a quick and straightforward way to use **alerts** as a Greenhouse Plugin on your Kubernetes cluster.

**Prerequisites**

- A running and Greenhouse-onboarded Kubernetes cluster. If you don't have one, follow the [Cluster onboarding](https://cloudoperators.github.io/greenhouse/docs/user-guides/cluster/onboarding) guide.
- kube-monitoring plugin (which brings in Prometheus Operator) **OR** stand alone: awareness to enable the deployment of Prometheus Operator with this plugin

**Step 1:**

> You can install the `alerts` package in your cluster by installing it with [Helm](https://helm.sh/docs/helm/helm_install) manually or let the Greenhouse platform lifecycle it for you automatically. For the latter, you can either:

Reviewer: ```suggestion
You can install the `alerts` package in your cluster with [Helm](https://helm.sh/docs/helm/helm_install) manually or let the Greenhouse platform lifecycle it for you automatically. For the latter, you can either:
```

- [Supernova](https://github.com/cloudoperators/juno/tree/main/apps/supernova)

Optionally, [Prometheus Operator](https://github.com/prometheus-operator/prometheus-operator) can be used to manage _Prometheus_ and _Alertmanager_ instances. Usually, **kube-monitoring** Plugin installs _Prometheus Operator_.

## Architecture

![Alerts Plugin Architecture](./img/Alerts-arch.png)

## Disclaimer

This is not meant to be a comprehensive package that covers all scenarios. If you are an expert, feel free to configure the plugin according to your needs.

The Plugin is a deeply configured [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/README.md) Helm chart which helps to keep track of versions and community updates.

It is intended as a platform that can be extended by following the [guide](#managing-alertmanager-configuration). 

Contribution is highly appreciated. If you discover bugs or want to add functionality to the plugin, then pull requests are always welcome.

## Quick start

This guide provides a quick and straightforward way to use **alerts** as a Greenhouse Plugin on your Kubernetes cluster.

**Prerequisites**

> - A running and Greenhouse-onboarded Kubernetes cluster. If you don't have one, follow the [Cluster onboarding](https://cloudoperators.github.io/greenhouse/docs/user-guides/cluster/onboarding) guide.

Reviewer: ```suggestion
- A running and Greenhouse-onboarded Kubernetes cluster. If you don't have one, follow the [Cluster onboarding](https://cloudoperators.github.io/greenhouse/docs/user-guides/cluster/onboarding) guide.
- kube-monitoring plugin (which brings in Prometheus Operator) **OR** stand alone: awareness to enable the deployment of Prometheus Operator with this plugin
```
